packages/oft-evm/test/mocks/NativeOFTAdapterMock.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { NativeOFTAdapter } from "../../contracts/NativeOFTAdapter.sol";
+
+contract NativeOFTAdapterMock is NativeOFTAdapter {
+    constructor(
+        uint8 _localDecimals,
+        address _lzEndpoint,
+        address _delegate
+    ) NativeOFTAdapter(_localDecimals, _lzEndpoint, _delegate) Ownable(_delegate) {}
+
+    // @dev expose internal functions for testing purposes
+    function debit(
+        uint256 _amountToSendLD,
+        uint256 _minAmountToCreditLD,
+        uint32 _dstEid
+    ) public returns (uint256 amountDebitedLD, uint256 amountToCreditLD) {
+        return _debit(msg.sender, _amountToSendLD, _minAmountToCreditLD, _dstEid);
+    }
+
+    function credit(address _to, uint256 _amountToCreditLD, uint32 _srcEid) public returns (uint256 amountReceivedLD) {
+        return _credit(_to, _amountToCreditLD, _srcEid);
+    }
```

Comment:
In general (so maybe in this case it's not applicable) testing internal functions is not the best idea. Internal functions can change as long as the external interface does not.

In these cases, I recommend testing these in the context of the external flow as they will never be run in isolation.

Check with @TRileySchwarz and @ryandgoulding though before changing anything, maybe there is some value to having them tested.

Reply:
I see what you mean, makes sense! I had originally done this to follow the [testing pattern of OFTAdapter](https://github.com/LayerZero-Labs/devtools/blob/555752ef24c787164cecd551aad363978da4d381/packages/oft-evm/test/OFT.t.sol#L329-L374), but if @TRileySchwarz and @ryandgoulding also agree, then I have a PR [here](https://github.com/LayerZero-Labs/devtools/pull/879) that removes testing internal functions. We already test the external flow in `test_native_oft_adapter_send()` so this PR just removes the internal function testing.

Reply:
Just makes it easier to encapsulated functions in a more controlled manner.  Since the contracts are immutable, it is generally pretty safe as any implementation change is likely going to go through re-audit, redeployment, etc.  But, it is fine to remove them if they are tested elsewhere.  In general, agreed, it is a shortcut to testing the API as it will actually be used.